### PR TITLE
Export Breadcrumbs Cron Job

### PR DESCRIPTION
**Pull request by ghost**
Cron Job to migrate LMS breadcrumbs from MongoDB to RDS to join to the LMS data.
-) Re-used the harvesting scripts from the original script
-) To order the data correctly based on the order in the LMS (instead of based on timestamp only) the script also retrieves the syllabus data from the Google sheet and joins it to the breadcrumbs
-) Used Pandas native .to_sql to upload the data to RDS for simplicity

### CODE REVIEW

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    """Retrieve the course syllabus from Google Sheet with the ordering
+
+    Returns a Dataframe to join to the rest of the breadcumbs
+    """
+    breadcrumb_index = requests.get(URL).json()
+    df_breadcrumb_idx = pd.DataFrame(breadcrumb_index['lessons'])
+    df_breadcrumb_idx = df_breadcrumb_idx.T
+    df_breadcrumb_idx = df_breadcrumb_idx.reset_index()
+    df_breadcrumb_idx.rename(columns={'index':'order_index'}, inplace=True)
+    # Temporary until this is being renamed in the LMS
```
Comment (yoniLavi) [inline lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py]: For comments like this, I would suggest the TODO prefix and a clear call to action, e.g.:
`# TODO: remove following line, once the [beta] suffix is removed in the LMS`

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        
+        # Looping through the results because you cannot convert 
+        # Variable length dict list to DataFrame, but list of lists
+        components = []
+        for k,v in all_components.iteritems():
+            temp = []
+            temp.append(k)
+            for item in v:
+                temp.append(item)
+            components.append(temp)
```
Comment (yoniLavi) [inline lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py]: This seems a bit too verbose. I would suggest something like (if I understand the data types correctly):
```
components = [[k] + v for k, v in d.iteritems()]
```

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+        df = pd.DataFrame(components)
+        # Need to assign headers
+        df.columns = ['uuid','module','section','lesson','unit','type','unknown_col']
```
Comment (yoniLavi) [inline lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py]: what is this 'unknown_col'? I would suggest a comment here

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        
+        # Need to get lesson order from syllabus for ordering the modules
+        # And not going back
+        df_breadcrumb_idx = get_breadcrumb_index(BREADCRUMB_INDEX_URL)
+        df = df.merge(df_breadcrumb_idx, on=['module', 'lesson'], how='left')
+        
+        engine = create_engine(CONNECTION_STRING, echo=False)
+        df.to_sql(name='lms_breadcrumbs_v2', con=engine, if_exists='replace',
```
Comment (yoniLavi) [inline lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py]: I'd suggest making a constant for `'lms_breadcrumbs_v2'` too

**Review by yoniLavi — CHANGES_REQUESTED** [review]
This is great, just minor comments

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py (hunk 1 above)
Comment (ghost) [reply lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py]: Amended in the subsequent commit

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py (hunk 2 above)
Comment (ghost) [reply lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py]: Changed the loop to list comprehension in the following commit

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py (hunk 3 above)
Comment (ghost) [reply lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py]: Removed column unknown_col. Seems to have been created when harvesting, but is empty and no use.

File: lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py (hunk 4 above)
Comment (ghost) [reply lms/djangoapps/learning_success/management/commands/export_all_breadcrumbs.py]: Created constant for the table name in following commit

**Review by yoniLavi — APPROVED** [review]
This is excellent

**Pull request merged**